static/js/utils/disaster_event_map_utils.tsx

```diff
@@ -339,55 +346,27 @@ function getDateRange(selectedDate: string): [string, string] {
 export function fetchDisasterEventPoints(
   dataOptions: DisasterDataOptions
 ): Promise<Record<string, DisasterEventPointData>> {
-  // Dates to fetch data for.
+  // Date range to fetch data for.
   const dateRange = getDateRange(dataOptions.selectedDate);
-  const dates = [];
-  const minYear = dateRange[0].substring(0, 4);
-  const maxYear = dateRange[1].substring(0, 4);
-  // The minimum length of a date string from dates in the dateRange and dates
-  // used for data retrieval. Dates used for data retrieval are YYYY-MM.
-  const minDateLength = Math.min(dateRange[0].length, 7);
-  // Loop through every YYYY-MM date between minYear-01 and maxYear-12 and add
-  // all dates that are within the dateRange.
-  for (let year = Number(minYear); year <= Number(maxYear); year++) {
-    for (let month = 1; month <= 12; month++) {
-      const date = `${year.toString()}-${month.toString().padStart(2, "0")}`;
-      if (
-        date.substring(0, minDateLength) <
-        dateRange[0].substring(0, minDateLength)
-      ) {
-        continue;
-      }
-      if (
-        date.substring(0, minDateLength) >
-        dateRange[1].substring(0, minDateLength)
-      ) {
-        break;
-      }
-      dates.push(date);
-    }
-  }
   const promises = [];
   // list of spec ids that correspond to the spec id used for the promise at
   // that index in the list of promises.
   const promiseSpecId = [];
   for (const eventSpec of dataOptions.eventTypeSpecs) {
     for (const eventType of eventSpec.eventTypeDcids) {
-      for (const date of dates) {
-        promiseSpecId.push(eventSpec.id);
-        promises.push(
-          fetchEventPoints(
-            eventType,
-            dataOptions.place,
-            date,
-            eventSpec.id,
-            dataOptions.severityFilters[eventSpec.id],
-            dateRange[0].length > 7 ? dateRange[0] : "",
-            dateRange[1].length > 7 ? dateRange[1] : "",
-            dataOptions.useCache
-          )
-        );
-      }
+      promiseSpecId.push(eventSpec.id);
+      promises.push(
+        fetchEventPoints(
+          eventType,
+          dataOptions.place,
+          dateRange,
+          eventSpec.id,
+          dataOptions.severityFilters[eventSpec.id],
+          dateRange[0].length > 7 ? dateRange[0] : "",
+          dateRange[1].length > 7 ? dateRange[1] : "",
```

Comment:
i don't understand the default value here. if dateRange is a year, we should still honor it. how about leaving that handling to the endpoint?

Reply:
ah, good catch, this was from old logic, removed it